From bb8eea9918febbe19ca9261b64bf494e1e9f945e Mon Sep 17 00:00:00 2001
From: GUO Zihua <guozihua@huawei.com>
Date: Tue, 9 Aug 2022 23:06:49 +0800
Subject: [PATCH] keys: Allow automatic module signature with SM3
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: bb8eea9918febbe19ca9261b64bf494e1e9f945e
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5K5HM
CVE: NA

--------------------------------

Support SM3 algorithm for module automatic signature during
"make modules_install" and "make modules_sign".

Signed-off-by: GUO Zihua <guozihua@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Reviewed-by: Chao Liu <liuchao173@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 init/Kconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 27c5ed16fef1..ea91c4864783 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2288,6 +2288,9 @@ config MODULE_SIG_SHA512
 	bool "Sign modules with SHA-512"
 	select CRYPTO_SHA512
 
+config MODULE_SIG_SM3
+	bool "Sign modules with SM3"
+	select CRYPTO_SM3
 endchoice
 
 config MODULE_SIG_HASH
@@ -2298,6 +2301,7 @@ config MODULE_SIG_HASH
 	default "sha256" if MODULE_SIG_SHA256
 	default "sha384" if MODULE_SIG_SHA384
 	default "sha512" if MODULE_SIG_SHA512
+	default "sm3" if MODULE_SIG_SM3
 
 config MODULE_COMPRESS
 	bool "Compress modules on installation"
-- 
2.33.0

